engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java

```diff
@@ -1460,6 +1466,11 @@ public void orchestrateStart(final String vmUuid, final Map<VirtualMachineProfil
         }
     }
 
+    private boolean checkForNonAllocatedVolumes(long vmId) {
+        final List<VolumeVO> vols = _volsDao.findByInstance(vmId);
+        return CollectionUtils.isEmpty(vols) || vols.stream().allMatch(v -> Volume.State.Allocated.equals(v.getState()));
```

Comment:
so false if any are not in  allocated-state? How does this logically is a `checkForNonAllocatedVolumes()`? seems like a `checkThatAllVolumesAreAllocated()` to me.